File: website/test_json.py
#!/usr/bin/env python3
import json
import os
from pathlib import Path

def test_scholars_json():
    """Test if the scholars.json file is valid and contains the expected data."""
    # Get the path to the scholars.json file
    file_path = Path('data/scholars.json')
    
    print(f"Testing {file_path}...")
    
    # Check if the file exists
    if not file_path.exists():
        print(f"ERROR: File not found: {file_path}")
        return False
    
    # Get the file size
    file_size = file_path.stat().st_size
    print(f"File size: {file_size} bytes")
    
    # Try to load the file
    try:
        with open(file_path, 'r') as f:
            scholars = json.load(f)
        
        # Check if it's a list
        if not isinstance(scholars, list):
            print(f"ERROR: Expected a list, got {type(scholars)}")
            return False
        
        # Check if it has items
        if len(scholars) == 0:
            print("ERROR: No scholars found in the file")
            return False
        
        print(f"Successfully loaded {len(scholars)} scholars")
        
        # Check the first scholar
        first_scholar = scholars[0]
        print(f"First scholar: {first_scholar.get('name', 'NO NAME')} (ID: {first_scholar.get('id', 'NO ID')})")
        
        # Check if scholars have the required fields
        valid_scholars = 0
        for i, scholar in enumerate(scholars):
            if i < 5 or i >= len(scholars) - 5:
                print(f"Scholar {i}: {scholar.get('name', 'NO NAME')} (ID: {scholar.get('id', 'NO ID')})")
            
            # Check required fields
            has_id = 'id' in scholar
            has_name = 'name' in scholar
            has_coords = 'coords' in scholar and isinstance(scholar['coords'], list) and len(scholar['coords']) == 2
            
            if has_id and has_name and has_coords:
                valid_scholars += 1
        
        print(f"Found {valid_scholars} valid scholars out of {len(scholars)}")
        
        return valid_scholars > 0
    
    except json.JSONDecodeError as e:
        print(f"ERROR: Invalid JSON: {e}")
        return False
    
    except Exception as e:
        print(f"ERROR: {e}")
        return False

if __name__ == "__main__":
    # Change to the script directory
    os.chdir(os.path.dirname(os.path.abspath(__file__)))
    
    # Test the scholars.json file
    if test_scholars_json():
        print("SUCCESS: scholars.json is valid")
    else:
        print("FAILURE: scholars.json is invalid") 

File: website/test_json.js
// Test script to load and process scholars.json
const fs = require('fs');

try {
    // Load the JSON file
    const data = fs.readFileSync('./data/scholars.json', 'utf8');
    console.log('File loaded successfully');
    
    // Parse the JSON
    const scholarsData = JSON.parse(data);
    console.log(`Number of scholars: ${Object.keys(scholarsData).length}`);
    
    // Convert to array format (same as in script.js)
    const scholars = Object.entries(scholarsData).map(([id, scholar]) => {
        return {
            id: scholar.id,
            name: scholar.name,
            institution: scholar.institution || '',
            country: scholar.country || '',
            umap: scholar.umap_projection ? [scholar.umap_projection.x, scholar.umap_projection.y] : null,
            cluster_id: scholar.cluster || 0
        };
    });
    
    console.log(`Converted ${scholars.length} scholars`);
    
    // Check for valid UMAP coordinates
    const validScholars = scholars.filter(scholar => {
        const hasUMAP = scholar.umap && Array.isArray(scholar.umap) && scholar.umap.length === 2;
        return scholar.id && scholar.name && hasUMAP;
    });
    
    console.log(`Found ${validScholars.length} valid scholars out of ${scholars.length}`);
    
    if (validScholars.length > 0) {
        console.log('Sample valid scholar:', validScholars[0]);
    }
    
} catch (error) {
    console.error('Error processing JSON:', error);
}
